challenge.rb

```diff
@@ -1,26 +1,26 @@
 
 def capitalize_each_string(input)
-  #implement your solution here
+  input.flat_map { |str| str.capitalize }
 end
 
 def fetch_the_dog(input)
-  #implement your solution here
+  input.find_all { |str| str.casecmp("dog") == 0 }
 end
 
 def no_dogs_allowed(input)
-  #implement your solution here
+  input.reject { |str| str.casecmp("dog") == 0 }
 end
 
 def count_the_animals(input)
-  #implement your solution here
+  input.count
 end
 
 def fetch_the_first_two(input)
-  #implement your solution here
+  input.first(2)
 end
 
 def fetch_CD_animals(input)
-  #implement your solution here
+  input.select { |str| str[0,1] == "c" or str[0,1] == "d"}
```

Comment:
Ruby strings have a handy [`start_with?`](http://ruby-doc.org/core-2.2.2/String.html#method-i-start_with-3F) method that should make this one read a bit better.